### PR TITLE
Sleepy ES5503 docs

### PR DESCRIPTION
**Pull request by LTVA1**
yep

### CODE REVIEW

**Comment by LTVA1** [comment]
Yes...

File: doc/4-instrument/es5503.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# ES5503 instrument editor
+
+ES5503 instrument editor is divided into 4 tabs:
```
Comment (ElectricKeet) [inline doc/4-instrument/es5503.md:3]:
```suggestion
the ES5503 instrument editor is divided into 4 tabs:
```

File: doc/4-instrument/es5503.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+- **ES5503**: for controlling the parameters of ES5503 sound channel.
+- **Wavetable**: for wavetable-related controls.
+- **Macros**: for macros controlling volume/arp/pitch/panning/etc.
+- **Sample**: for sample-related controls.
+
+## ES5503
+
+There is a tricky feature of ES5503 when it comes to playing wavetables/samples. ES5503 can play waves of sizes 256, 512, ..., 32768 steps. When it encounters `0` in sample memory it stops playing the wavetable/sample, no matter what oscillator mode and wave size are.
```
Comment (ElectricKeet) [inline doc/4-instrument/es5503.md:12]: Perhaps "waves of sizes that are multiples of 256"? Either that or add "768" to the sequence to make it clear.

File: doc/4-instrument/es5503.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+- **Oscillator mode**: lets you select one of four possible oscillator modes:
+  - **Freerun**: oscillator plays the whole wavetable/sample and loops back if it haven't encountered `0` in sample memory along the way. So if sample does not exactly fit to the one of aforementioned possible sizes it will stop playing after being played once.
+  - **Oneshot**: oscillator plays the whole wavetable/sample once.
+  - **Sync/AM**: sync mode is only possible between pairs of even and odd numbered oscillators, i.e. a lower even-numbered oscillator paired with a higher odd-numbered oscillator. When the even-numbered oscillator starts playing back its wavetable, its odd-numbered mate will be synchronized to it and begins its wavetable/sample simultaneously (aka hard sync). (shamelessly stolen from [here](http://www.buchty.net/ensoniq/5503.html)) AM means amplitude modulation and works in reverse - odd-numbered oscillator modulates the volume of previous even-numbered oscillator. It's kinda tricky to work with this mode.
```
Comment (ElectricKeet) [inline doc/4-instrument/es5503.md:17]:
```suggestion
  - **Sync/AM**: sync mode is only possible between pairs of even and odd numbered oscillators, i.e. a lower even-numbered oscillator paired with a higher odd-numbered oscillator. When the even-numbered oscillator starts playing back its wavetable, its odd-numbered mate will be synchronized to it and begins its wavetable/sample simultaneously, aka hard sync. (info from [www.buchty.net](http://www.buchty.net/ensoniq/5503.html) AM means amplitude modulation and works in reverse; odd-numbered oscillator modulates the volume of previous even-numbered oscillator. It's kinda tricky to work with this mode.
```

I don't want to strip the personality from everything, but "shamelessly stolen" might be a bit much. :)

File: doc/4-instrument/es5503.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  - **Swap**: uses even/odd pairs of oscillators as explained above. Enabled oscillator plays its wavetable/sample once, then stops as if in oneshot mode, and launches its mate. This is how it is possible to play 64 KiB sample seamlessly.
+- **Virtual softpan channel**: this is a software feature. Under the hood is just uses a pair of odd and even numbered channels, assigns them with the same wave, oscillator mode and frequency. Thus you effectively have one channel with 256-step panning. Furnace provides macros for individual left and right panning control.
+- **Phase reset on key-on**: resets phase of the channel each time new note plays. Does not apply when `03xx` and `EAxx` effects are used on the same row.
```
Comment (ElectricKeet) [inline doc/4-instrument/es5503.md:20]:
```suggestion
- **Phase reset on key-on**: resets phase of the channel each time a new note plays. Does not apply when `03xx` and `EAxx` effects are used on the same row.
```

File: doc/7-systems/README.md [hunk 5]
```diff
@@ -70,6 +70,7 @@ this is the full list of chips that Furnace supports.
 - [MOS 6581/8580 (SID)](c64.md)
 - [Ensoniq ES5506](es5506.md)
 - [ESS ES1xxx-series (ESFM)](esfm.md)
+- [ESS ES1xxx-series (ESFM)](es5503.md)
```
Comment (ElectricKeet) [inline doc/7-systems/README.md:73]:
```suggestion
- [ES5503](es5503.md)
```

File: doc/7-systems/es5503.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+# Ensoniq ES5503 (ES5503)
+
+A sound chip designed by Robert "Bob" Yannes (the same guy who designed the [MOS Technology SID](c64.md)). It features 32 time-multiplexed channels (oscillators) of wavetable/sample audio.
```
Comment (ElectricKeet) [inline doc/7-systems/es5503.md:3]: Just for consistency's sake with other documentation, perhaps start this and the following sentences with lowercase. (It's a weird quirk of tilde's writing style, but I'm not about to go through and "correct" everything to proper sentence case, so lowercase it is.)

File: doc/7-systems/es5503.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+It was used in some Ensoniq synthesizers and in Apple IIGS computer.
+
+Chip has 8 audio output channels, but each oscillator can be connected to only one output. Furnace maps them left-right-left-right-... by default. There's an option to downmix chip output to mono audio (as it was on original Apple IIGS).
+
+Chip allows limited inter-channel effects (AM and hard sync).
```
Comment (ElectricKeet) [inline doc/7-systems/es5503.md:9]:
```suggestion
the chip allows limited inter-channel effects (AM and hard sync).
```

File: doc/7-systems/es5503.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
+# Ensoniq ES5503 (ES5503)
+
+A sound chip designed by Robert "Bob" Yannes (the same guy who designed the [MOS Technology SID](c64.md)). It features 32 time-multiplexed channels (oscillators) of wavetable/sample audio.
+
+It was used in some Ensoniq synthesizers and in Apple IIGS computer.
+
+Chip has 8 audio output channels, but each oscillator can be connected to only one output. Furnace maps them left-right-left-right-... by default. There's an option to downmix chip output to mono audio (as it was on original Apple IIGS).
```
Comment (ElectricKeet) [inline doc/7-systems/es5503.md:7]:
```suggestion
the chip has 8 audio output channels, but each oscillator can be connected to only one output. Furnace maps them left-right-left-right-... by default. there's an option to downmix chip output to mono audio (as it was on original Apple IIGS).
```

**Review by ElectricKeet — CHANGES_REQUESTED** [review]
Thank you for your patience with all these suggestions!

File: doc/4-instrument/es5503.md (hunk 2 above)
Comment (LTVA1) [reply doc/4-instrument/es5503.md:12]: I should have continued... It's 256, 512, 1024, 2048, 4096, 8192, 16384 and 32768

**Review by ElectricKeet — APPROVED** [review]
Looks good to me!

**Pull request merged**